src/pages/studyView/UserSelections.tsx

```diff
@@ -439,9 +444,75 @@ export default class UserSelections extends React.Component<
                 </div>
             );
         }
+
+        if (this.props.filter.sampleTreatmentFilters) {
+            const f = this.renderTreatmentFilter(this.props.filter.sampleTreatmentFilters);
+            components.push(f)
+        }
+
+        if (this.props.filter.patientTreatmentFilters) {
+            const f = this.renderTreatmentFilter(this.props.filter.patientTreatmentFilters);
+            components.push(f);
+        }
         return components;
     }
 
+    private renderTreatmentFilter(f: AndedPatientTreatmentFilters | AndedSampleTreatmentFilters): JSX.Element {
+        type OuterFilter = OredPatientTreatmentFilters | OredSampleTreatmentFilters;
+        type InnerFilter = PatientTreatmentFilter | SampleTreatmentFilter;
+
+        // the gross "as any" casting shouldn't be necessary for a type of
+        // OredSampleTreatmentFilters[] | OredPatientTreatmentFilters[],
+        // but the compiler complains if I remove it
```

Comment:
I've run into this typing issue before. When you have an array of ambiguous type `A[] | B[]` typescript doesnt know how to resolve the methods for typing. To make it work it would have to be `(A|B)[]` instead.

Reply:
TIL. Good to know, thank you.